marimo/_islands/island_generator.py

```diff
@@ -380,6 +376,46 @@ def render_body(
         for stub in self._stubs:
             rendered_stubs.append(stub.render())
 
+        if include_initialization_indicator:
+            dummy_cell_id = self._app.cell_manager.create_cell_id()
+            dummy_input = "<marimo-cell-code hidden> '' </marimo-cell-code>"
+            dummy_output = dedent("""
+            <div class="marimo" style="--tw-bg-opacity: 0;">
+              <div class="flex flex-col items-center justify-center">
+                <svg
+                  xmlns="http://www.w3.org/2000/svg"
+                  width="24"
+                  height="24"
+                  viewBox="0 0 24 24"
+                  fill="none"
+                  stroke="currentColor"
+                  stroke-width="1"
+                  stroke-linecap="round"
+                  stroke-linejoin="round"
+                  class="size-20 animate-spin text-primary"
+                >
+                  <path d="M21 12a9 9 0 1 1-6.219-8.56"></path>
+                </svg>
+                <div>Initializing...</div>
+              </div>
+            </div>
+            """)
+            dummy_stub = dedent(
+                f"""
+                <marimo-island
+                    data-app-id="{self._app_id}"
+                    data-cell-id="{dummy_cell_id}"
+                    data-reactive="{json.dumps(True)}"
+                >
+                    <marimo-cell-output>
+                    {dummy_output}
+                    </marimo-cell-output>
+                    {dummy_input}
+                </marimo-island>
+                """
+            )
+            rendered_stubs = [dummy_stub] + rendered_stubs
```

Comment:
Not really clear why you are changing the order? Shouldn't you be adding dummy stub with real code input? instead of both?